src/map/mod.rs

```diff
@@ -116,6 +122,46 @@ impl PositionType {
     }
 }
 
+#[pyclass(eq, eq_int, frozen, hash)]
+#[derive(Default, PartialEq, Debug, Clone, Copy, Hash)]
+pub(super) enum RotationAngle {
+    #[pyo3(name = "DEG_0")]
+    #[default]
+    Deg0 = 0,
+    #[pyo3(name = "DEG_90")]
+    Deg90 = 90,
+    #[pyo3(name = "DEG_180")]
+    Deg180 = 180,
+    #[pyo3(name = "DEG_270")]
+    Deg270 = 270,
+}
+
+impl From<i16> for RotationAngle {
+    fn from(value: i16) -> Self {
+        match value {
+            0 => RotationAngle::Deg0,
+            90 => RotationAngle::Deg90,
+            180 => RotationAngle::Deg180,
+            270 => RotationAngle::Deg270,
+            _ => {
+                warn!(
+                    "Invalid rotation angle: {}. Valid values are 0, 90, 180, 270. Defaulting to 0.",
+                    value
+                );
+                RotationAngle::Deg0
+            }
+        }
+    }
+}
+
+#[pymethods]
+impl RotationAngle {
+    #[staticmethod]
+    fn from_int(value: i16) -> Self {
+        RotationAngle::from(value)
```

Comment:
Please move the mapping from above here and raise instead of creating a log entry.

The reason behind this is that I had a discussion today with a colleague about it and it's easier to add new functionality instead of providing the wrong one (Using 0 instead the correct angle)

I don't expect that different values currently exist but we will see if users are reporting errors

Reply:
Sure, done